hazelcast-client/src/test/java/com/hazelcast/client/ClientRegressionTest.java

```diff
@@ -887,6 +887,55 @@ public void run() {
 
         hazelcastInstance.shutdown();
 
+        assertOpenEventually("Put operations should not hang.", testFinishedLatch);
+    }
+
+
+    @Test
+    public void testClusterShutdown_thenCheckOperationsNotHanging_withGet() throws Exception {
+        HazelcastInstance hazelcastInstance = hazelcastFactory.newHazelcastInstance();
+
+        ClientConfig clientConfig = new ClientConfig();
+        //Retry all requests
+        clientConfig.getNetworkConfig().setRedoOperation(true);
+        //Retry all requests forever(until client is shutdown)
+        clientConfig.setProperty(ClientProperties.PROP_INVOCATION_TIMEOUT_SECONDS, String.valueOf(Integer.MAX_VALUE));
+        HazelcastInstance client = hazelcastFactory.newHazelcastClient(clientConfig);
+
+        final IMap<Object, Object> map = client.getMap(randomMapName());
+        final int mapSize = 1000;
+
+        final CountDownLatch clientStartedDoingRequests = new CountDownLatch(1);
```

Comment:
should the latch value be equal to the number of threads?

Reply:
I think one is enough, should not make much difference. The test is not new, it is directly copied from later commits because one of the backported pr was fixing this test see https://github.com/hazelcast/hazelcast/pull/7635, https://github.com/hazelcast/hazelcast/issues/7582